framework/db/BaseActiveRecord.php

```diff
@@ -606,15 +606,19 @@ public function markAttributeDirty($name)
      */
     public function isAttributeChanged($name, $identical = true)
     {
-        if (isset($this->_attributes[$name], $this->_oldAttributes[$name])) {
-            if ($identical) {
-                return $this->_attributes[$name] !== $this->_oldAttributes[$name];
-            }
+        if (!array_key_exists($name, $this->_attributes)) {
+            return false;
```

Comment:
This adds a new interpretation to missing keys in `_attributes`.
I don't think that's needed.

Reply:
I did it on purpose to satisfy the tests I wrote below, I might have missed something.
Could you describe a use case where a missing attribute (in ``_attributes``) should be considered as changed/dirty?

Reply:
I'd have to look into it a bit more, but I don't see why this new interpretation is needed at all.